### PR TITLE
new package: py-netket

### PR DESCRIPTION
**Pull request by Sinan81**

### CODE REVIEW

**Comment by Sinan81** [comment]
flake8 failing for unrelated packages but not this PR.

File: var/spack/repos/builtin/packages/py-netket/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+class PyNetket(PythonPackage):
+    """
+    NetKet is an open-source project, delivering cutting-edge methods for the
+    study of many-body quantum systems with artificial neural networks and
+    machine learning techniques.
+    """
+
+    homepage = "https://www.example.com"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netket/package.py]:
```suggestion
    homepage = "https://github.com/netket/netket"
```

File: var/spack/repos/builtin/packages/py-netket/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    version('2.1',   sha256='041d2b058c5d2945bf0a4194ad2bf3c426ad9b6ce0dd323a81a7154bc6e45452')
+    version('2.0',   sha256='c2890361b16ffb5265023a736536c435ccb3ad956d897e5820eac431d72cdb0e')
+    version('1.0.5', sha256='26562bf608775f21eb2cb443f2c66c09cadb56e6eac84f4d855d62e7d776a511')
+    version('1.0.4', sha256='0b344d526ee34d187281d0c2f7952c91728abbe22553e3dbcd45fcfeb312c3b5')
+    version('1.0.3', sha256='b8e54d7ad8b379b740def640d748c6560943aed473755389fc5cf1020b9007de')
+    version('1.0.2', sha256='229c906e92a432bbbd0ff0527874f41318f8fc480d12a33c8184f30960ae628b')
+
+    # build only deps
+    depends_on('py-setuptools', type='build')
+    depends_on('cmake', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netket/package.py]:
```suggestion
    depends_on('py-cmake@3.12:', type='build')
```

File: var/spack/repos/builtin/packages/py-netket/package.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    depends_on('blas')
+    depends_on('mpi')
+    depends_on('python', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netket/package.py]:
```suggestion
    depends_on('python@3.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-netket/package.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-scipy', type=('build', 'run'))
+    depends_on('py-mpi4py', type=('build', 'run'))
+    depends_on('py-numba', type=('build', 'run'))
+    depends_on('py-tqdm', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netket/package.py]: Missing version requirements:
https://github.com/netket/netket/blob/master/setup.py#L177,L184

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-netket/package.py (hunk 4 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-netket/package.py]: Thanks!

File: var/spack/repos/builtin/packages/py-netket/package.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    version('2.1',   sha256='041d2b058c5d2945bf0a4194ad2bf3c426ad9b6ce0dd323a81a7154bc6e45452')
+    version('2.0',   sha256='c2890361b16ffb5265023a736536c435ccb3ad956d897e5820eac431d72cdb0e')
+    version('1.0.5', sha256='26562bf608775f21eb2cb443f2c66c09cadb56e6eac84f4d855d62e7d776a511')
+    version('1.0.4', sha256='0b344d526ee34d187281d0c2f7952c91728abbe22553e3dbcd45fcfeb312c3b5')
+    version('1.0.3', sha256='b8e54d7ad8b379b740def640d748c6560943aed473755389fc5cf1020b9007de')
+    version('1.0.2', sha256='229c906e92a432bbbd0ff0527874f41318f8fc480d12a33c8184f30960ae628b')
+
+    # build only deps
+    depends_on('py-setuptools', type='build')
+    depends_on('cmake@3.12:', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netket/package.py]:
```suggestion
    depends_on('py-cmake@3.12:', type='build')
```
You may need to add an additional package for this dependency.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-netket/package.py (hunk 5 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-netket/package.py]: The package seemed to compile and run fine without it. Should we leave it as TODO?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-netket/package.py]: It's likely downloading the dependency itself via setuptools, that won't work on air-gapped clusters.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**